src/main/java/com/diffplug/gradle/eclipse/MavenCentralMapping.java

```diff
@@ -79,20 +83,30 @@ public static String groupIdArtifactId(String bundleId) {
 			if ("artifact".equals(artifact.getNodeName())) {
 				String classifier = artifact.getAttributes().getNamedItem("classifier").getNodeValue();
 				if ("osgi.bundle".equals(classifier)) {
-					String id = artifact.getAttributes().getNamedItem("id").getNodeValue();
-					String version = artifact.getAttributes().getNamedItem("version").getNodeValue();
-					Version parsed = Version.parseVersion(version);
-					map.put(id, parsed.getMajor() + "." + parsed.getMinor() + "." + parsed.getMicro());
+					String bundleId = artifact.getAttributes().getNamedItem("id").getNodeValue();
+					String bundleVersion = artifact.getAttributes().getNamedItem("version").getNodeValue();
+					String groupIdArtifactId = groupIdArtifactId(bundleId);
+					String version = calculateMavenCentralVersion(bundleId, bundleVersion);
+					map.put(groupIdArtifactId, version);
 				}
 			}
 		}
 		return map;
 	}
 
+	static String calculateMavenCentralVersion(String bundleId, String bundleVersion) {
+		Version parsed = Version.parseVersion(bundleVersion);
+		if (ICU_BUNDLE_ID.equals(bundleId) && parsed.getMicro() == 0) {
+			return parsed.getMajor() + "." + parsed.getMinor();
+		} else {
+			return parsed.getMajor() + "." + parsed.getMinor() + "." + parsed.getMicro();
+		}
+	}
+
 	private static final String ARTIFACTS_JAR = "artifacts.jar";
 
-	/** Returns a map from every bundle-id to its corresponding 3-part version (the qualifier is dropped). */
-	public static Map<String, String> bundleToVersion(EclipseRelease release) {
```

Comment:
This PR is good except that it nukes this `public static` method. 100% fine to just have two maps with duplicate info, I don't care. But I think there is at least one other project out there which is using this `bundleToVersion` method which would get broken by this change.

Reply:
I took your "it is OK to break previous behavior" from https://github.com/diffplug/goomph/issues/167#issuecomment-957981919 a little bit to strong :-)

I was not aware of projects potentially re-using the methods of the gradle plugin, so I am glad I also renamed the method. It makes it obvious that the keys have changed in the map.

I will restore the old public method, restore some of the unit tests, so that this changes stays backward compatible.

Reply:
I took your _"it is OK to break previous behavior"_ from https://github.com/diffplug/goomph/issues/167#issuecomment-957981919 a little bit too seriously :-)

I was not aware of projects potentially re-using the methods of the gradle plugin, so I am glad I also renamed the method. It makes it obvious that the keys have changed in the map.

I will restore the old public method, restore some of the unit tests, so that this changes stays backward compatible.

Reply:
Lol, that's my bad, changing this method was a very reasonable interpretation of my comment 🤣

Reply:
I have pushed c486db1b5d208efc65d182f3c2cc05a4f15beeeb to solve this